zkevm-circuits/src/mpt_circuit.rs

```diff
@@ -285,502 +385,233 @@ impl<F: FieldExt> MPTConfig<F> {
+                }}
+                // ACCOUNT_LEAF_NONCE_BALANCE_S
+                // ACCOUNT_LEAF_NONCE_BALANCE_C
+                let account_leaf_nonce_balance_s;
+                let account_leaf_nonce_balance_c;
+                ifx!{f!(position_cols.q_not_first) => {
+                    ifx!{a!(account_leaf.is_nonce_balance_s) => {
+                        account_leaf_nonce_balance_s = AccountLeafNonceBalanceConfig::configure(meta, &mut cb, ctx.clone(), true);
+                    }}
+                    ifx!{a!(account_leaf.is_nonce_balance_c) => {
+                        account_leaf_nonce_balance_c = AccountLeafNonceBalanceConfig::configure(meta, &mut cb, ctx.clone(), false);
+                    }}
+                }}
+                // ACCOUNT_LEAF_STORAGE_CODEHASH_S
+                // ACCOUNT_LEAF_STORAGE_CODEHASH_C
+                let account_leaf_storage_codehash_s = AccountLeafStorageCodehashConfig::configure(meta, &mut cb, ctx.clone(), true);
+                let account_leaf_storage_codehash_c = AccountLeafStorageCodehashConfig::configure(meta, &mut cb, ctx.clone(), false);
+                // ACCOUNT_DRIFTED_LEAF
+                let account_leaf_key_in_added_branch;
+                ifx!{f!(position_cols.q_not_first), a!(position_cols.not_first_level), a!(account_leaf.is_in_added_branch) => {
+                    account_leaf_key_in_added_branch = AccountLeafKeyInAddedBranchConfig::configure(meta, &mut cb, ctx.clone());
+                }}
+
+                /* Range checks */
+                // These range checks ensure that the value in the RLP columns are all byte value.
+                // These lookups also enforce the value to be zero if the passed in length is < 0.
+                // TODO(Brecht): would be safer/cleaner if this can be enabled everywhere even for branch child rlp1
+                // TODO(Brecht): do 2 bytes/lookup when circuit height >= 2**21
+                ifx!{f!(position_cols.q_enable) => {
+                    // Sanity checks (can be removed, here for safety)
+                    require!(cb.length_s.sum_conditions() => bool);
+                    require!(cb.length_c.sum_conditions() => bool);
+                    // Range checks
+                    ifx!{not!(a!(branch.is_child)) => {
+                        for &byte in ctx.rlp_bytes()[0..1].into_iter() {
+                            require!((FixedTableTag::RangeKeyLen256, a!(byte), 0.expr()) => @"fixed");
+                        }
+                    }}
+                    for &byte in ctx.rlp_bytes()[1..2].into_iter() {
+                        require!((FixedTableTag::RangeKeyLen256, a!(byte), 0.expr()) => @"fixed");
+                    }
+                    for (idx, &byte) in ctx.rlp_bytes()[2..34].into_iter().enumerate() {
+                        require!((cb.get_range_s(), a!(byte), cb.get_length_s() - (idx + 1).expr()) => @"fixed");
```

Comment:
This is so much nicer than it's in my code!!

I would just add some short description, like: "Checking there are zeros after the stream ends. For example if the leaf value length is 3, we need to make sure there are zeros in the unused columns."